src/coreclr/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -55,36 +55,39 @@ private IDictionary CreateDataContainer()
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern bool IsImmutableAgileException(Exception e);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern IRuntimeMethodInfo GetMethodFromStackTrace(object stackTrace);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ExceptionNative_GetMethodFromStackTrace")]
+        private static partial void GetMethodFromStackTrace(ObjectHandleOnStack stackTrace, ObjectHandleOnStack method);
 
         private MethodBase? GetExceptionMethodFromStackTrace()
         {
-            Debug.Assert(_stackTrace != null, "_stackTrace shouldn't be null when this method is called");
-            IRuntimeMethodInfo method = GetMethodFromStackTrace(_stackTrace!);
+            if (_stackTrace == null)
+            {
+                return null;
+            }
+
+            object stackTraceLocal = _stackTrace;
+            IRuntimeMethodInfo? methodInfo = null;
+            GetMethodFromStackTrace(ObjectHandleOnStack.Create(ref stackTraceLocal), ObjectHandleOnStack.Create(ref methodInfo));
+
+            // Ensure the stack trace lives beyond the current above call.
+            // See implementation for details on stacktrace data.
+            GC.KeepAlive(stackTraceLocal);
```

Comment:
Doesn't `ObjectHandleOnStack` keep the object alive, if the reference isn't written in unmanaged code?

And also, can the `object[]`/`sbyte[]` handling be in managed?

Reply:
I agree - GC.KeepAlive should not be needed with the new structure of the code.

Reply:
Ugh. I forgot to remove this. Yes, this should be removed.